flow/connectors/eventhub/hub_batches.go

```diff
@@ -79,6 +84,71 @@ func (h *HubBatches) ForEach(fn func(ScopedEventhub, *azeventhubs.EventDataBatch
 	}
 }
 
+func (h *HubBatches) sendBatch(
+	ctx context.Context,
+	tblName ScopedEventhub,
+	events *azeventhubs.EventDataBatch,
+) error {
+	subCtx, cancel := context.WithTimeout(ctx, 5*time.Minute)
+	defer cancel()
+
+	hub, err := h.manager.GetOrCreateHubClient(subCtx, tblName)
+	if err != nil {
+		return err
+	}
+
+	opts := &azeventhubs.SendEventDataBatchOptions{}
+	err = hub.SendEventDataBatch(subCtx, events, opts)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("successfully sent %d events to event hub topic - %s", events.NumEvents(), tblName.ToString())
+	return nil
+}
+
+func (h *HubBatches) flushAllBatches(
+	ctx context.Context,
+	maxParallelism int64,
+	flowName string,
+) error {
+	if h.Len() == 0 {
+		log.WithFields(log.Fields{
+			"flowName": flowName,
+		}).Infof("no events to send")
+		return nil
+	}
+
+	var numEventsPushed int32
+	g, gCtx := errgroup.WithContext(ctx)
+	g.SetLimit(int(maxParallelism)) // limit parallel merges to 8
+	tableNameRowsMapping := cmap.New[uint32]()
+	h.ForEach(func(tblName ScopedEventhub, eventBatch *azeventhubs.EventDataBatch) {
+		g.Go(func() error {
+			numEvents := eventBatch.NumEvents()
+			err := h.sendBatch(gCtx, tblName, eventBatch)
+			if err != nil {
+				return err
+			}
+
+			atomic.AddInt32(&numEventsPushed, numEvents)
+			log.WithFields(log.Fields{
+				"flowName": flowName,
+			}).Infof("pushed %d events to event hub: %s", numEvents, tblName)
+			rowCount, ok := tableNameRowsMapping.Get(tblName.ToString())
+			if !ok {
+				rowCount = uint32(0)
+			}
+			rowCount += uint32(numEvents)
+			tableNameRowsMapping.Set(tblName.ToString(), rowCount)
```

Comment:
This isn't safe from races. If two threads read rowCount before either sets, they'll overwrite each other

Use Upsert: https://github.com/orcaman/concurrent-map/blob/v2.0.1/concurrent_map.go#L85